process/track/blockProcessor.go

```diff
@@ -336,6 +336,16 @@ func (bp *blockProcessor) checkHeaderFinality(
 			continue
 		}
 
+		// if the currentHeader(the one that should confirm the finality of the prev)
```

Comment:
probably not needed if we handle inside the block tracker the check that there is a proof.
Will take the change of block tracker in a separate PR.

Can you add a check somewhere on the processing (for both leader and other nodes - in create block for leader & process block for others) that every used block has proof

Reply:
this should already be covered:
- create block
   - shard: https://github.com/multiversx/mx-chain-go/blob/fix_transition_block/process/block/shardblock.go#L1975
   - meta: https://github.com/multiversx/mx-chain-go/blob/fix_transition_block/process/block/metablock.go#L1162
- process block:
   - shard: https://github.com/multiversx/mx-chain-go/blob/fix_transition_block/process/block/shardblock.go#L581
   - meta: https://github.com/multiversx/mx-chain-go/blob/fix_transition_block/process/block/metablock.go#L2026

Reply:
when verifying the proof for the header not the prev proof, we should *not* use `common.IsFlagEnabledAfterEpochsStartBlock(header, sp.enableEpochsHandler, common.EquivalentMessagesFlag)`

so I think `func (sp *shardProcessor) checkMetaHdrFinality(header data.HeaderHandler) error` has the wrong check